### ISSUE TITLE
Exception ''package:flutter/src/rendering/semantics.dart': Failed assertion: line 674: 'node.parent ?. _dirty != true' is not true.' on ios.

### ISSUE THREAD

**Issue by sigurdm** [flutter/flutter#7287]
## Steps to Reproduce

Run this program (known to fail on the ios simulator simulating iphone 6s ios 9.2) and tap the first item.
```
import 'package:flutter/material.dart';

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(title: new Text("foo")),
        body: new Block(children: <Widget>[
          new MaterialList(children: <Widget>[
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
            new _GalleryItem(),
          ])
        ]));
  }
}

class _GalleryItem extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new ListItem(
        title: new Text("foo"),
        subtitle: new Text("foo"),
        onTap: () {
          Navigator.push(
              context, new MaterialPageRoute(builder: (_) => new Text("Foo")));
        });
  }
}

void main() {
  runApp(new MaterialApp(title: "foo", home: new Home()));
}
```

The error seems to only appear when the list is longer than the screen.

I have not been able to reproduce this on android.

## Logs
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
'package:flutter/src/rendering/semantics.dart': Failed assertion: line 674 pos 14:
'node.parent?._dirty != true' is not true.
When the exception was thrown, this was the stack:
#2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/rendering/semantics.dart:674:14)
#3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1089:23)
#4      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.beginFrame (package:flutter/src/rendering/binding.dart:224:19)
#5      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.beginFrame (package:flutter/src/widgets/binding.dart:314:23)
#6      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:165:5)
#7      BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:653:15)
#8      BindingBase&SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:588:9)
#9      _beginFrame (file:///b/build/slave/Mac_Engine/build/src/flutter/lib/ui/hooks.dart:83)
(elided 2 frames from class _AssertionError)
════════════════════════════════════════════════════════════════════════════════════════════════════

## Flutter Doctor

$ flutter doctor
[✓] Flutter (on Mac OS, channel master)
    • Flutter at /Users/sigurdm/flutter
    • Framework revision 1b9939af95 (33 hours ago), 2016-12-17 18:14:37
    • Engine revision 2f32e8054d
    • Tools Dart version 1.21.0

[x] Android toolchain - develop for Android devices
    x Android Studio / Android SDK not found. Download from https://developer.android.com/sdk/
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] iOS toolchain - develop for iOS devices (Xcode 8.1)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.1, Build version 8B62

[-] IntelliJ IDEA Community Edition (version 2016.2)
    • Dart plugin version 162.1469
    • Flutter plugin not installed; this adds Flutter specific functionality.
    • For information about managing plugins, see
      https://www.jetbrains.com/help/idea/2016.2/managing-plugins.html

[✓] Connected devices
    • iPhone 6s • C1A09776-279F-43CC-B763-9D66C2259A59 • ios • iOS 9.2 (simulator)

**Comment by eseidelGoogle** [comment]
I feel like we've seen this before.

**Comment by eseidelGoogle** [comment]
@Hixie likely knows this part of the AX code.

**Comment by goderbauer** [comment]
I cannot reproduce this after adaption the example code to the new widget names. Also, this code has changed a lot since this bug was filed. Closing for now. Feel free to file a new bug if you still run into this.
